app/collins/solr/SolrHelper.scala

```diff
@@ -111,35 +111,30 @@ object SolrHelper {
     }.getOrElse(logger.warn("attempted to populate solr when no server was initialized"))
   }
 
-  def updateItems[T](items: Seq[T], serializer: SolrSerializer[T], indexTime: Date, commit: Boolean = true) {
+  def updateItems[T](items: Seq[T], serializer: SolrSerializer[T], indexTime: Date) = {
     _server.map { server =>
       val docs = items.map { item => prepForInsertion(serializer.serialize(item, indexTime)) }
       if (docs.size > 0) {
         val fuckingJava = new java.util.ArrayList[SolrInputDocument]
         docs.foreach { doc => fuckingJava.add(doc) }
-        server.add(fuckingJava)
-        if (commit) {
-          server.commit()
-          if (items.size == 1) {
-            logger.debug(("Indexed %s: %s".format(serializer.docType.name.toLowerCase, items.head.toString)))
-          } else {
-            logger.info("Indexed %d %ss".format(docs.size, serializer.docType.name.toLowerCase))
-          }
-        }
+        server.add(fuckingJava, SolrConfig.commitWithin)
+        logger.debug("Added %d %s documents to be indexed within %d ms".format(
+          fuckingJava.size,
+          serializer.docType.name.toLowerCase,
+          SolrConfig.commitWithin))
+        // dont explicitly hard commit, let solr figure it out and make docs available
```

Comment:
no, because a hard commit isnt actually synchronous because this is happening via actor in background thread